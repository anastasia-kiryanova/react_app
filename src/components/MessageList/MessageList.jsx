import React from 'react'
import PropTypes from 'prop-types'
import styles from './MessageList.module.css'

export const MessageList = ({ messages }) => {
  return (
    <ul className={styles.messsagesList}>
      {messages.map(message => 
        <li className={styles.messageItem} key={message.id}>{message.author} : {message.text}</li>
      )}
    </ul>
  )
}

MessageList.propTypes = {
  messages: PropTypes.array.isRequired
}