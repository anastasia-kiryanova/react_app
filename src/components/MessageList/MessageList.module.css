.messageItem {
  list-style-type: none;
  padding: 15px;
  background-color: rgb(247, 247, 247);
  border: 1px solid rgb(189, 189, 189);
  width: 500px;
  margin-bottom: 15px;
}

.messageItem:nth-child(2n) {
  color: rgb(52, 52, 255);
  background-color: rgb(209, 209, 209);
}

.messsagesList {
  margin: 0;
  padding: 0;
}
