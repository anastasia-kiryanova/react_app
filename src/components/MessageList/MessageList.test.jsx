import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { MessageList } from './MessageList';

describe ('MessageList', () => {
  it ('is a Component', () => {
    expect(MessageList).toBeInstanceOf(Function);
  });
  
  it ('test snapshot', () => {
    const mess = [
      {
        id: 1,
        author: 'test',
        text: 'test message'
      }
    ];
    const { asFragment } = render(<MessageList messages={mess} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('message list is empty', () => {
    const mess = [];
    render(<MessageList messages={mess} />);
    expect(screen.getByRole('list')).toBeTruthy();
    expect(screen.queryByRole('listitem')).toBeFalsy();
  });

  it('render message', () => {
    const mess = [
      {
        id: 1,
        author: 'test',
        text: 'test message'
      }
    ];
    render(<MessageList messages={mess} />);
    expect(screen.getByRole('listitem')).toBeTruthy();
    expect(screen.getByText(/test/)).toBeTruthy();
    expect(screen.getByText(/test message/)).toBeTruthy();
  });  
});