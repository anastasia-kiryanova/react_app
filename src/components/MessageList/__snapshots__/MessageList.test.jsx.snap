// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList test snapshot 1`] = `
<DocumentFragment>
  <ul
    class="messsagesList"
  >
    <li
      class="messageItem"
    >
      test : test message
    </li>
  </ul>
</DocumentFragment>
`;
